src/tools/compiletest/src/lib.rs

```diff
@@ -800,10 +802,10 @@ fn collect_tests_from_dir(
 // Added to allow reuse by custom Ferrocene code
 fn find_tests_in_dir(
     config: Arc<Config>,
-    dir: &Path,
-    relative_dir_path: &Path,
-    modified_tests: &Vec<PathBuf>,
-    on_test_found: &mut dyn FnMut(&TestPaths, Option<PathBuf>),
+    dir: &Utf8Path,
+    relative_dir_path: &Utf8Path,
+    modified_tests: &Vec<Utf8PathBuf>,
+    on_test_found: &mut dyn FnMut(&TestPaths, Option<Utf8PathBuf>),
```

Comment:
why change these types

Reply:
I found those changes necessary to make the file compile

Reply:
This change was introduced upstream on https://github.com/rust-lang/rust/commit/e24b0c8e0a9d0b898ede993b62c32d9ebaff27bd